contracts/GenesisPass.vy

```diff
@@ -67,114 +85,270 @@ def __init__(_initialMintWallet: address):
 
 
 @external
-def safeTransferFrom(_from: address, _to: address, _tokenId: uint256, _data: Bytes[1024]=b""):
+def safeTransferFrom(_from: address, _to: address, _tokenId: uint256, data: Bytes[1024]=b""):
+
     """
-    @notice Not supported
+    @notice Transfers the ownership of an NFT from one address to another address
+    @dev Throws unless `msg.sender` is the current owner, an authorized operator, or the approved address for this NFT. Throws if `_from` is
+     not the current owner. Throws if `_to` is the zero address. Throws if `_tokenId` is not a valid NFT. When transfer is complete, this function
+     checks if `_to` is a smart contract (code size > 0). If so, it calls `onERC721Received` on `_to` and throws if the return value is not
+     `bytes4(keccak256("onERC721Received(address,address,uint256,bytes)"))`.
+    @param _from The current owner of the NFT
+    @param _to The new owner
+    @param _tokenId The NFT to transfer
+    @param data Optional additional data with no specified format, sent in call to `_to`
     """
-    raise "not supported"
+
+    self._transferFrom(_from, _to, _tokenId)
+    if _to.is_contract:
+        returnValue: bytes4 = ERC721Receiver(_to).onERC721Received(msg.sender, _from, _tokenId, data)
+        assert returnValue == method_id("onERC721Received(address,address,uint256,bytes)", output_type=bytes4), "transfer to non-ERC721Receiver"
+
 
 @external
 def approve(_approved: address, _tokenId: uint256):
+
     """
-    @notice Not supported
+    @notice Change or reaffirm the approved address for an NFT
+    @dev The zero address indicates there is no approved address.
+     Throws unless `msg.sender` is the current NFT owner, or an authorized operator of the current owner.
+    @param _approved The new approved NFT controller
+    @param _tokenId The NFT to approve
     """
-    raise "not supported"
+
+    _owner: address = self.tokenOwner[_tokenId]
+
+    assert _approved != _owner, "approval to current owner"
+    assert msg.sender == _owner or self.isApprovedForAll[_owner][msg.sender], "caller not owner nor operator"
+
+    self.tokenApprovals[_tokenId] = _approved
+    log Approval(_owner, _approved, _tokenId)
 
 
 @external
 def setApprovalForAll(_operator: address, _approved: bool):
+
     """
-    @notice Not supported
+    @notice Enable or disable approval for a third party ("operator") to manage all of `msg.sender`'s assets
+    @dev Emits the ApprovalForAll event. The contract MUST allow multiple operators per owner.
+    @param _operator Address to add to the set of authorized operators
+    @param _approved True if the operator is approved, false to revoke approval
     """
-    raise "not supported"
+
+    assert msg.sender != _operator, "approve to sender"
+
+    self.isApprovedForAll[msg.sender][_operator] = _approved
+    log ApprovalForAll(msg.sender, _operator, _approved)
 
 
 @external
-def mint(_to: address, _tokenId: uint256) -> bool:
+def mint(_to: address) -> uint256:
```

Comment:
The goal is to have the minting of the total supply of passes in the constructor. After that, there can't be more NFTs minted. Therefore, this method should do nothing and the `INITIAL_SUPPLY` variable should actually be something like `TOTAL_SUPPLY`